src/targets/gpu/hip.cpp

```diff
@@ -47,14 +47,22 @@ MIGRAPHX_REGISTER_OP(hip_copy_literal)
 using hip_ptr      = MIGRAPHX_MANAGE_PTR(void, hipFree);
 using hip_host_ptr = MIGRAPHX_MANAGE_PTR(void, hipHostUnregister);
 
-std::string hip_error(int error) { return hipGetErrorString(static_cast<hipError_t>(error)); }
+std::string hip_get_and_reset_error()
+{
+    // reset HIP's internal status to hipSuccess
+    auto last_error = hipGetLastError();
+    return std::string(hipGetErrorString(static_cast<hipError_t>(last_error)));
+}
 
 bool is_device_ptr(const void* ptr)
 {
     hipPointerAttribute_t attr;
     auto status = hipPointerGetAttributes(&attr, ptr);
     if(status != hipSuccess)
+    {
+        (void)(hip_get_and_reset_error());
```

Comment:
We should only be calling `hipLastError` for launching the kernel since that is the only call which doesnt return an error.